src/test/java/no/entur/antu/validator/servicelinksvalidator/ServiceLinksValidatorTest.java

```diff
@@ -0,0 +1,456 @@
+package no.entur.antu.validator.servicelinksvalidator;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+import java.util.Optional;
+import net.opengis.gml._3.DirectPositionType;
+import no.entur.antu.commondata.CommonDataRepository;
+import no.entur.antu.model.QuayId;
+import no.entur.antu.model.StopPlaceCoordinates;
+import no.entur.antu.netextestdata.NetexTestData;
+import no.entur.antu.stop.StopPlaceRepository;
+import no.entur.antu.validator.ValidationContextWithNetexEntitiesIndex;
+import org.entur.netex.index.api.NetexEntitiesIndex;
+import org.entur.netex.validation.validator.ValidationReport;
+import org.entur.netex.validation.validator.ValidationReportEntry;
+import org.entur.netex.validation.validator.ValidationReportEntrySeverity;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.rutebanken.netex.model.ServiceLink;
+
+class ServiceLinksValidatorTest {
```

Comment:
add test with journey pattern without service links and service link with invalid geometry.